module/platform/NUSense/HardwareIO/src/HardwareIO.cpp

```diff
@@ -85,11 +86,67 @@ namespace module::platform::NUSense {
             // Tell the stream reactor to connect to the device
             emit(std::make_unique<ConnectSerial>(device, baud));
 
+            // TODO: Apply the servo offsets within NUSense instead
             // Apply servo offsets
             for (size_t i = 0; i < config["servos"].config.size(); ++i) {
                 nugus.servo_offset[i]    = config["servos"][i]["offset"].as<Expression>();
                 nugus.servo_direction[i] = config["servos"][i]["direction"].as<Expression>();
+
+                cfg.servo_configurations[i].direction = config["servos"][i]["direction"].as<int32_t>();
+                cfg.servo_configurations[i].offset = config["servos"][i]["offset"].as<double>();
             }
+
+            handshake_watchdog =
+                on<Watchdog<HandshakeWatchdog, 1, std::chrono::seconds>, Sync<HandshakeWatchdog>>().then(
+                    "Sending handshake message to NUSense",
+                    [this] {
+                        // Parse handshake packet
+                        auto handshake = NUSenseHandshake();
+
+                        handshake.msg = "";
+
+                        for (size_t i = 0; i < cfg.servo_configurations.size(); ++i) {
+                            handshake.servo_configs.emplace_back(cfg.servo_configurations[i].direction,
+                                                                 cfg.servo_configurations[i].offset);
+                        }
+
+                        // Send to stream reactor
+                        send_packet(handshake);
+
+                        log<INFO>("Handshake message sent to NUSense, waiting for response...");
+                    });
+
+            // Sync is used because uart write is a shared resource
+            NUC_packet_catcher =
+                on<Trigger<ServoTargets>, Sync<HardwareIO>>().then([this](const ServoTargets& commands) {
+                    // Copy the data into a new message so we can use a duration instead of a timepoint
+                    // and take the offsets and switch the direction.
+                    auto servo_targets = SubcontrollerServoTargets();
+
+                    // Change the timestamp in servo targets to the difference between the timestamp and now
+                    // Take away the offset and switch the direction if needed
+                    for (auto& target : commands.targets) {
+                        servo_targets.targets.emplace_back(
+                            target.time - NUClear::clock::now(),
+                            target.id,
+                            (target.position - nugus.servo_offset[target.id]) * nugus.servo_direction[target.id],
```

Comment:
If you are going to send the offsets and the directions in the handshake message and deal with them on the subcontroller-side -- which is the intended vision --, then you should remove them here.